Merchant/index.ts

```diff
@@ -46,7 +46,9 @@ export namespace Merchant {
 			isoly.CountryCode.Alpha2.is(value.country) &&
 			typeof value.categoryCode == "string" &&
 			MerchantRules.is(value.rules) &&
-			(value.processor == undefined || MerchantProcessor.is(value.processor))
+			(value.processor == undefined ||
+				MerchantProcessor.is(value.processor) ||
+				(MerchantProcessor.is(value.processor.visa) && MerchantProcessor.is(value.processor.mastercard)))
```

Comment:
You are accessing a property (value.processor) that may not even be an object, which could result in a crash if it is a string or simular.
```suggestion
				(typeof value.processor == "object" && value.processor &&
				MerchantProcessor.is(value.processor.visa) && MerchantProcessor.is(value.processor.mastercard)))
```